File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

const resistor_band_t expected[] = {
BLACK, BROWN, RED, ORANGE, YELLOW,
GREEN, BLUE, VIOLET, GREY, WHITE
};
int color_code(resistor_band_t r){
	return (int)r;
}

const resistor_band_t* colors(){
   return expected;
}

File: kotlin/resistor-color/src/main/kotlin/ResistorColor.kt
object ResistorColor {

    private val colorMap: Map<String, Int> = colors().withIndex().associate { (i,s) -> Pair(s, i) }.toMap()

    fun colorCode(input: String): Int {
        return colorMap.getOrDefault(input, -1)
    }

    fun colors(): List<String> {
        return listOf<String>("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white")
    }

}
